File: CPP/coolakt/345. Reverse Vowels of a String.cpp
class Solution {
public:
    string reverseVowels(string s) {
        string t = "";
        for (int i = 0; i < s.size(); i++) {
            if (s[i] == 'a' || s[i] == 'e' || s[i] == 'i' || s[i] == 'o' || s[i] == 'u' ||
                s[i] == 'A' || s[i] == 'E' || s[i] == 'I' || s[i] == 'O' || s[i] == 'U') {
                t.push_back(s[i]);
            }
        }

        int l = t.size() - 1;
        for (int j = 0; j < s.size(); j++) {
            if (s[j] == 'a' || s[j] == 'e' || s[j] == 'i' || s[j] == 'o' || s[j] == 'u' ||
                s[j] == 'A' || s[j] == 'E' || s[j] == 'I' || s[j] == 'O' || s[j] == 'U') {
                s[j] = t[l];
                l--;
            }
        }
        return s;
    }
};

File: Python/kumarAbhi-kumar/345_Reverse_Vowels_of_A_String.py
class Solution(object):
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        s_list = list(s)
        len_s_list = len(s_list)
        left = 0
        right = len_s_list - 1
        v_list = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']
        while left < right:
            while left < len_s_list and s_list[left] not in v_list:
                left += 1
            while right >= 0 and s_list[right] not in v_list:
                right -= 1
            if left < right:
                s_list[left], s_list[right] = s_list[right], s_list[left]
                left += 1
                right -= 1
        s = ''.join(s_list)
        return s